src/engine/sched.c

```diff
@@ -1646,13 +1670,12 @@ sched_start_cycle(struct sched_data *data, ABT_pool *pools)
 	if (sched_relax_mode != SCHED_RELAX_MODE_DISABLED)
 		sched_try_relax(dx, pools, cycle->sc_ults_tot);
 
-	if (sched_stats_intvl != 0 &&
-	    (info->si_stats.ss_print_ts + sched_stats_intvl) <
-	    info->si_cur_ts) {
-		D_PRINT("XS(%d) CPU time(ms): Total:"DF_U64", Relax:"DF_U64"\n",
-			dx->dx_xs_id, info->si_stats.ss_tot_time,
-			info->si_stats.ss_relax_time);
-		info->si_stats.ss_print_ts = info->si_cur_ts;
+	d_tm_inc_counter(info->si_stats.ss_total_time, duration);
+	d_tm_set_counter(info->si_stats.ss_wq_len, info->si_req_cnt);
+	d_tm_set_counter(info->si_stats.ss_sq_len, info->si_sleep_cnt);
```

Comment:
I know this has already landed, but for the record I believe this is an incorrect use of the counter type. Counters should always increase (unless they've wrapped around). Downstream consumers expect this behavior and may be confused by a counter that is varying up and down. In this case, I believe that the queue length should be represented by a gauge.

Reply:
Oh, I wasn't aware of this, and I think there are other places where we use counter in this way, maybe I'd push a follow-on patch to fix all these counters.